### PR TITLE
mongodb-7_0: init at 7.0.14

### PR DESCRIPTION
**Pull request by qubitnano**
## Description of changes

https://www.mongodb.com/docs/manual/release-notes/7.0/

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by qubitnano** [comment]
Result of `nixpkgs-review pr 335665` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>3 packages built:</summary>
  <ul>
    <li>mongodb (mongodb-6_0)</li>
    <li>mongodb-5_0</li>
    <li>mongodb-7_0</li>
  </ul>
</details>

**Comment by drupol** [comment]
Closes #333258

Nice!!! How long it took you to build it? I've been trying many things in here: https://github.com/NixOS/nixpkgs/pull/333258

**Comment by qubitnano** [comment]
> Closes #333258
> 
> Nice!!! How long it took you to build it? I've been trying many things in here: #333258

About an hour and a half on my machine. Ryzen 3700X. mongodb-6_0 takes about an hour. I couldn't build with tmpfs even with 64GB.

**Comment by drupol** [comment]
Sorry, I rephrase my question.

I've been trying to build version 7.0.12 last week and spend almost a day trying to figure out how to fix the builds. I was just wondering how long it took you to figure it out.

File: pkgs/servers/nosql/mongodb/mongodb.nix [hunk 1]
```diff
@@ -48,15 +48,17 @@ let
...
     "zlib"
     #"asio" -- XXX use package?
     #"stemmer"  -- not nice to package yet (no versioning, no makefile, no shared libs).
     #"valgrind" -- mongodb only requires valgrind.h, which is vendored in the source.
     #"wiredtiger"
-  ] ++ lib.optionals stdenv.isLinux [ "tcmalloc" ];
+  ] ++ lib.optionals stdenv.isLinux [ "tcmalloc" ]
+    ++ lib.optionals (lib.versionOlder version "7.0") [
+      "pcre"
+    ];
```
Comment (emilazy) [inline pkgs/servers/nosql/mongodb/mongodb.nix:61]: Is our PCRE too old or too new or something?

**Comment by drupol** [comment]
We should also find a way to provide binary builds of this. Does the licence prevent us to do so at all?

**Comment by emilazy** [comment]
That’s not something the project has consensus for at present, see https://github.com/NixOS/nixpkgs/issues/83884 (my personal opinion at https://github.com/NixOS/nixpkgs/issues/83884#issuecomment-2241222950; SSPL is an especially egregious licence and I would personally consider it much less safe for us than many other `unfreeRedistributable` licences).

External projects are of course welcome to take that risk themselves and provide their own build and cache infrastructure (https://github.com/numtide/nixpkgs-unfree used to do this).

File: pkgs/servers/nosql/mongodb/mongodb.nix (hunk 1 above)
Comment (qubitnano) [reply pkgs/servers/nosql/mongodb/mongodb.nix:61]: --use-system-pcre was removed, maybe pcre2 will work

Comment (qubitnano) [reply pkgs/servers/nosql/mongodb/mongodb.nix]: seems to build fine with pcre2

**Comment by emilazy** [comment]
By the way, we could probably avoid the patch by just including `lld` in `nativeBuildInputs`?

**Comment by drupol** [comment]
Can we temporarily disable the licence so we can see how it builds on all the platforms?
Once we see it, we can put it back as it was.

**Comment by markuskowa** [comment]
Result of `nixpkgs-review pr 335665 --extra-nixpkgs-config '{allowUnfree = true;}'` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>3 packages built:</summary>
  <ul>
    <li>mongodb (mongodb-6_0)</li>
    <li>mongodb-5_0</li>
    <li>mongodb-7_0</li>
  </ul>
</details>

**Comment by emilazy** [comment]
> Can we temporarily disable the licence so we can see how it builds on all the platforms? Once we see it, we can put it back as it was.

Please don’t do this; we shouldn’t try to circumvent ofborg’s policy, especially for things with legal implications. If we want it to test `unfreeRedistributable` packages we should change that in ofborg.

**Comment by GaetanLepage** [comment]
Result of `nixpkgs-review pr 335665` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package marked as broken and skipped:</summary>
  <ul>
    <li>mongodb-5_0</li>
  </ul>
</details>
<details>
  <summary>2 packages failed to build:</summary>
  <ul>
    <li>mongodb (mongodb-6_0)</li>
    <li>mongodb-7_0</li>
  </ul>
</details>

**Comment by GaetanLepage** [comment]
Result of `nixpkgs-review pr 335665` run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>mongodb-5_0</li>
  </ul>
</details>
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>mongodb (mongodb-6_0)</li>
    <li>mongodb-7_0</li>
  </ul>
</details>

**Comment by drupol** [comment]
@GaetanLepage Thank you! Can you link the build logs for Darwin (the failures). Thanks again!

**Comment by GaetanLepage** [comment]
> @GaetanLepage Thank you! Can you link the build logs for Darwin (the failures). Thanks again!

Sure. They all fail in this manner on `darwin`:
```
Running phase: buildPhase
install: skipping file '/dev/fd/63', as it was replaced while being copied
scons build flags: -j8 --release --ssl --wiredtiger=on --js-engine=mozjs --use-sasl-client --disable-warnings-as-errors VARIANT_DIR=nixos --link-model=static MONGO_VERSION=6.0.16 --use-system-boost --use-system-snappy --use-system-yaml --use-system-zlib --use-system-pcre CC=clang CXX=clang++ CCFLAGS=\'-march=armv8-a+crc\' DESTDIR=/nix/store/dalkhxs2b3x8p35kjl86h09nf43kf86j-mongodb-6.0.16
scons: Reading SConscript files ...
PermissionError: [Errno 1] Operation not permitted: '/etc/paths.d':
  File "/private/tmp/nix-build-mongodb-6.0.16.drv-0/source/SConstruct", line 25:
    DefaultEnvironment(tools=[])
  File "/nix/store/qkfl8dz1vq564m9ibnqpzmg77lmq1ad6-scons-4.7.0/lib/python3.11/site-packages/SCons/Defaults.py", line 82:
    _default_env = SCons.Environment.Environment(*args, **kwargs)
  File "/nix/store/qkfl8dz1vq564m9ibnqpzmg77lmq1ad6-scons-4.7.0/lib/python3.11/site-packages/SCons/Environment.py", line 1217:
    platform(self)
  File "/nix/store/qkfl8dz1vq564m9ibnqpzmg77lmq1ad6-scons-4.7.0/lib/python3.11/site-packages/SCons/Platform/__init__.py", line 138:
    return self.generate(*args, **kw)
  File "/nix/store/qkfl8dz1vq564m9ibnqpzmg77lmq1ad6-scons-4.7.0/lib/python3.11/site-packages/SCons/Platform/darwin.py", line 48:
    dirlist = os.listdir('/etc/paths.d')
```

**Comment by sikmir** [comment]
> > @GaetanLepage Thank you! Can you link the build logs for Darwin (the failures). Thanks again!
> 
> Sure. They all fail in this manner on `darwin`:
> 
> ```
> Running phase: buildPhase
> install: skipping file '/dev/fd/63', as it was replaced while being copied
> scons build flags: -j8 --release --ssl --wiredtiger=on --js-engine=mozjs --use-sasl-client --disable-warnings-as-errors VARIANT_DIR=nixos --link-model=static MONGO_VERSION=6.0.16 --use-system-boost --use-system-snappy --use-system-yaml --use-system-zlib --use-system-pcre CC=clang CXX=clang++ CCFLAGS=\'-march=armv8-a+crc\' DESTDIR=/nix/store/dalkhxs2b3x8p35kjl86h09nf43kf86j-mongodb-6.0.16
> scons: Reading SConscript files ...
> PermissionError: [Errno 1] Operation not permitted: '/etc/paths.d':
>   File "/private/tmp/nix-build-mongodb-6.0.16.drv-0/source/SConstruct", line 25:
>     DefaultEnvironment(tools=[])
>   File "/nix/store/qkfl8dz1vq564m9ibnqpzmg77lmq1ad6-scons-4.7.0/lib/python3.11/site-packages/SCons/Defaults.py", line 82:
>     _default_env = SCons.Environment.Environment(*args, **kwargs)
>   File "/nix/store/qkfl8dz1vq564m9ibnqpzmg77lmq1ad6-scons-4.7.0/lib/python3.11/site-packages/SCons/Environment.py", line 1217:
>     platform(self)
>   File "/nix/store/qkfl8dz1vq564m9ibnqpzmg77lmq1ad6-scons-4.7.0/lib/python3.11/site-packages/SCons/Platform/__init__.py", line 138:
>     return self.generate(*args, **kw)
>   File "/nix/store/qkfl8dz1vq564m9ibnqpzmg77lmq1ad6-scons-4.7.0/lib/python3.11/site-packages/SCons/Platform/darwin.py", line 48:
>     dirlist = os.listdir('/etc/paths.d')
> ```

The same with all other scons based packages on darwin:
```sh
mapnik> scons: Reading SConscript files ...
mapnik> PermissionError: [Errno 1] Operation not permitted: '/etc/paths.d':
mapnik>   File "/private/tmp/nix-build-mapnik-4.0.2.drv-0/source/SConstruct", line 142:
mapnik>     env = Environment(ENV=os.environ)
mapnik>   File "/nix/store/h055rwhjfjkvdh1x4xcgaq30cwfdjggj-scons-4.7.0/lib/python3.12/site-packages/SCons/Environment.py", line 1217:
mapnik>     platform(self)
mapnik>   File "/nix/store/h055rwhjfjkvdh1x4xcgaq30cwfdjggj-scons-4.7.0/lib/python3.12/site-packages/SCons/Platform/__init__.py", line 138:
mapnik>     return self.generate(*args, **kw)
mapnik>   File "/nix/store/h055rwhjfjkvdh1x4xcgaq30cwfdjggj-scons-4.7.0/lib/python3.12/site-packages/SCons/Platform/darwin.py", line 48:
mapnik>     dirlist = os.listdir('/etc/paths.d')
```

```sh
hammer-nightly_20220416> scons: Reading SConscript files ...
hammer-nightly_20220416> PermissionError: [Errno 1] Operation not permitted: '/etc/paths.d':
hammer-nightly_20220416>   File "/private/tmp/nix-build-hammer-nightly_20220416.drv-0/source/SConstruct", line 31:
hammer-nightly_20220416>     env = Environment(ENV = envvars,
hammer-nightly_20220416>   File "/nix/store/h055rwhjfjkvdh1x4xcgaq30cwfdjggj-scons-4.7.0/lib/python3.12/site-packages/SCons/Environment.py", line 1217:
hammer-nightly_20220416>     platform(self)
hammer-nightly_20220416>   File "/nix/store/h055rwhjfjkvdh1x4xcgaq30cwfdjggj-scons-4.7.0/lib/python3.12/site-packages/SCons/Platform/__init__.py", line 138:
hammer-nightly_20220416>     return self.generate(*args, **kw)
hammer-nightly_20220416>   File "/nix/store/h055rwhjfjkvdh1x4xcgaq30cwfdjggj-scons-4.7.0/lib/python3.12/site-packages/SCons/Platform/darwin.py", line 48:
hammer-nightly_20220416>     dirlist = os.listdir('/etc/paths.d')
```

**Comment by al3xtjames** [comment]
I ran into the SCons issue with an unrelated package while doing a large rebuild. The following patch should fix it:

```patch
diff --git a/SCons/Platform/darwin.py b/SCons/Platform/darwin.py
index 381b54249..4ab7466c6 100644
--- a/SCons/Platform/darwin.py
+++ b/SCons/Platform/darwin.py
@@ -46,7 +46,7 @@ def generate(env) -> None:
     # make sure this works on Macs with Tiger or earlier
     try:
         dirlist = os.listdir('/etc/paths.d')
-    except FileNotFoundError:
+    except (FileNotFoundError, PermissionError):
         dirlist = []
 
     for file in dirlist:

```

**Comment by GaetanLepage** [comment]
Result of `nixpkgs-review pr 335665` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages marked as broken and skipped:</summary>
  <ul>
    <li>mongodb</li>
    <li>mongodb-5_0</li>
    <li>mongodb-6_0</li>
    <li>mongodb-7_0</li>
  </ul>
</details>

**Comment by GaetanLepage** [comment]
Result of `nixpkgs-review pr 335665` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages marked as broken and skipped:</summary>
  <ul>
    <li>mongodb</li>
    <li>mongodb-5_0</li>
    <li>mongodb-6_0</li>
    <li>mongodb-7_0</li>
  </ul>
</details>

**Comment by GaetanLepage** [comment]
Result of `nixpkgs-review pr 335665` run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>mongodb-5_0</li>
  </ul>
</details>
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>mongodb (mongodb-6_0)</li>
    <li>mongodb-7_0</li>
  </ul>
</details>

**Comment by GaetanLepage** [comment]
Result of `nixpkgs-review pr 335665` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>3 packages built:</summary>
  <ul>
    <li>mongodb (mongodb-6_0)</li>
    <li>mongodb-5_0</li>
    <li>mongodb-7_0</li>
  </ul>
</details>

**Comment by GaetanLepage** [comment]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

- Command: `nixpkgs-review pr 335665`
- Systems: `x86_64-linux`,`aarch64-linux`,`x86_64-darwin`,`aarch64-darwin`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 3 packages built:</summary>
  <ul>
    <li>mongodb (mongodb-6_0)</li>
    <li>mongodb-5_0</li>
    <li>mongodb-7_0</li>
  </ul>
</details>

---
### `aarch64-linux`
<details>
  <summary>:x: 1 package failed to build:</summary>
  <ul>
    <li>mongodb-5_0</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 2 packages built:</summary>
  <ul>
    <li>mongodb (mongodb-6_0)</li>
    <li>mongodb-7_0</li>
  </ul>
</details>

---
### `x86_64-darwin`
<details>
  <summary>:fast_forward: 4 packages marked as broken and skipped:</summary>
  <ul>
    <li>mongodb</li>
    <li>mongodb-5_0</li>
    <li>mongodb-6_0</li>
    <li>mongodb-7_0</li>
  </ul>
</details>

---
### `aarch64-darwin`
<details>
  <summary>:fast_forward: 4 packages marked as broken and skipped:</summary>
  <ul>
    <li>mongodb</li>
    <li>mongodb-5_0</li>
    <li>mongodb-6_0</li>
    <li>mongodb-7_0</li>
  </ul>
</details>

**Comment by qubitnano** [comment]
I reverted marking all of Darwin as broken. Scons itself needs patched, likely targeting staging.

**Comment by drupol** [comment]
@GaetanLepage Can you trigger a build once again?

**Comment by GaetanLepage** [comment]
> @GaetanLepage Can you trigger a build once again?

Sure, building right now !

**Comment by GaetanLepage** [comment]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 335665`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 3 packages built:</summary>
  <ul>
    <li>mongodb (mongodb-6_0)</li>
    <li>mongodb-5_0</li>
    <li>mongodb-7_0</li>
  </ul>
</details>

---
### `aarch64-linux`
<details>
  <summary>:x: 1 package failed to build:</summary>
  <ul>
    <li>mongodb-5_0</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 2 packages built:</summary>
  <ul>
    <li>mongodb (mongodb-6_0)</li>
    <li>mongodb-7_0</li>
  </ul>
</details>

---
### `x86_64-darwin`
<details>
  <summary>:fast_forward: 4 packages marked as broken and skipped:</summary>
  <ul>
    <li>mongodb</li>
    <li>mongodb-5_0</li>
    <li>mongodb-6_0</li>
    <li>mongodb-7_0</li>
  </ul>
</details>

---
### `aarch64-darwin`
<details>
  <summary>:fast_forward: 4 packages marked as broken and skipped:</summary>
  <ul>
    <li>mongodb</li>
    <li>mongodb-5_0</li>
    <li>mongodb-6_0</li>
    <li>mongodb-7_0</li>
  </ul>
</details>

**Comment by GaetanLepage** [comment]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 335665`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 3 packages built:</summary>
  <ul>
    <li>mongodb (mongodb-6_0)</li>
    <li>mongodb-5_0</li>
    <li>mongodb-7_0</li>
  </ul>
</details>

---
### `aarch64-linux`
<details>
  <summary>:x: 1 package failed to build:</summary>
  <ul>
    <li>mongodb-5_0</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 2 packages built:</summary>
  <ul>
    <li>mongodb (mongodb-6_0)</li>
    <li>mongodb-7_0</li>
  </ul>
</details>

---
### `x86_64-darwin`
<details>
  <summary>:fast_forward: 4 packages marked as broken and skipped:</summary>
  <ul>
    <li>mongodb</li>
    <li>mongodb-5_0</li>
    <li>mongodb-6_0</li>
    <li>mongodb-7_0</li>
  </ul>
</details>

---
### `aarch64-darwin`
<details>
  <summary>:fast_forward: 4 packages marked as broken and skipped:</summary>
  <ul>
    <li>mongodb</li>
    <li>mongodb-5_0</li>
    <li>mongodb-6_0</li>
    <li>mongodb-7_0</li>
  </ul>
</details>

**Comment by drupol** [comment]
thank you !

**Pull request merged**